build error when building 32bit libraries on x86_64 host

Hi,

while updating the 32bit version of util-linux libraries I encounter the following build error:
```
[607/608] Compiling C object enosys.p/misc-utils_enosys.c.o
FAILED: enosys.p/misc-utils_enosys.c.o 
ccache cc -Ienosys.p -I. -I../util-linux-2.40 -Iinclude -I../util-linux-2.40/include -Ilibblkid -I../util-linux-2.40/libblkid -I../util-linux-2.40/libblkid/src -Ilibsmartcols -I../util-linux-2.40/libsmartcols -I../util-linux-2.40/libsmartcols/src -Ilibmount -I../util-linux-2.40/libmount -I../util-linux-2.40/libmount/src -Ilibfdisk -I../util-linux-2.40/libfdisk -I../util-linux-2.40/libfdisk/src -I../util-linux-2.40/libuuid/src -I../util-linux-2.40/liblastlog2/src -Isys-utils -I../util-linux-2.40/sys-utils -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -D_GNU_SOURCE -include /home/pkgmk/work/util-linux-32/src/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -D_TIME_BITS=64 -O2 -march=x86-64 -m32 -fPIE -MD -MQ enosys.p/misc-utils_enosys.c.o -MF enosys.p/misc-utils_enosys.c.o.d -o enosys.p/misc-utils_enosys.c.o -c ../util-linux-2.40/misc-utils/enosys.c
distcc[17721] ERROR: compile ../util-linux-2.40/misc-utils/enosys.c on 10.6.66.30/14 failed
distcc[17721] (dcc_build_somewhere) Warning: remote compilation of '../util-linux-2.40/misc-utils/enosys.c' failed, retrying locally
distcc[17721] Warning: failed to distribute ../util-linux-2.40/misc-utils/enosys.c to 10.6.66.30/14, running locally instead
distcc[17721] ERROR: compile ../util-linux-2.40/misc-utils/enosys.c on localhost failed
./syscalls.h:2:22: error: '__NR_accept' undeclared here (not in a function)
    2 | UL_SYSCALL("accept", __NR_accept)
      |                      ^~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:40:29: error: '__NR_epoll_ctl_old' undeclared here (not in a function)
   40 | UL_SYSCALL("epoll_ctl_old", __NR_epoll_ctl_old)
      |                             ^~~~~~~~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:44:30: error: '__NR_epoll_wait_old' undeclared here (not in a function)
   44 | UL_SYSCALL("epoll_wait_old", __NR_epoll_wait_old)
      |                              ^~~~~~~~~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:133:31: error: '__NR_kexec_file_load' undeclared here (not in a function)
  133 | UL_SYSCALL("kexec_file_load", __NR_kexec_file_load)
      |                               ^~~~~~~~~~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:190:26: error: '__NR_newfstatat' undeclared here (not in a function)
  190 | UL_SYSCALL("newfstatat", __NR_newfstatat)
      |                          ^~~~~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:268:24: error: '__NR_security' undeclared here (not in a function)
  268 | UL_SYSCALL("security", __NR_security)
      |                        ^~~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:272:21: error: '__NR_semop' undeclared here (not in a function)
  272 | UL_SYSCALL("semop", __NR_semop)
      |                     ^~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:273:26: error: '__NR_semtimedop' undeclared here (not in a function)
  273 | UL_SYSCALL("semtimedop", __NR_semtimedop)
      |                          ^~~~~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
./syscalls.h:340:23: error: '__NR_tuxcall' undeclared here (not in a function)
  340 | UL_SYSCALL("tuxcall", __NR_tuxcall)
      |                       ^~~~~~~~~~~~
../util-linux-2.40/misc-utils/enosys.c:71:38: note: in definition of macro 'UL_SYSCALL'
   71 | #define UL_SYSCALL(name, nr) { name, nr },
      |                                      ^~
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /home/pkgmk/work/util-linux-32/src/build -j 44
```

Attached is a full [build.log](https://github.com/util-linux/util-linux/files/14812458/build.log)

--- a/meson.build
+++ b/meson.build
@@ -2708,7 +2708,7 @@ endif
 errnos_h = custom_target('errnos.h',
   input : 'tools/all_errnos',
   output : 'errnos.h',
-  command : ['tools/all_errnos', cc.cmd_array()]
+  command : ['tools/all_errnos', cc.cmd_array(), get_option('c_args')],
 )
 
 mq_libs = []
@@ -3026,7 +3026,7 @@ endif
 syscalls_h = custom_target('syscalls.h',
   input : 'tools/all_syscalls',
   output : 'syscalls.h',
-  command : ['tools/all_syscalls', cc.cmd_array()] 
+  command : ['tools/all_syscalls', cc.cmd_array(), get_option('c_args')],
 )
 
 if cc.compiles(fs.read('include/audit-arch.h'), name : 'has AUDIT_ARCH_NATIVE')

